### ISSUE TITLE
Coral_loss during training is 0

### ISSUE THREAD

**Issue by arnancjlx** [jindongwang/transferlearning#97]
Hi, Wang, thx for your code, it really helps me a lot.
I run your code but the coral_loss during training is 0 all the time. Is this because of the implementation of the DeepCoral loss part is wrong?

**Comment by jindongwang** [comment]
Other users have also used the code, and they are fine. Could you please re-run the code?

**Comment by arnancjlx** [comment]
Thx for your reply, I've re-run the code for several times but the result still be 0. The only change I did is setting the batch_size to 8 due to my cuda memory. Could that be the reason?

**Comment by jindongwang** [comment]
It is hard to know why... You can try other batches such as batch_size=6 / 4 to see if it works fine.

I'll find some time to run the code by myself to check again.

**Comment by arnancjlx** [comment]
thank you so much!

**Comment by jindongwang** [comment]
I update the code a few days ago, you can check if this still happens

**Comment by arnancjlx** [comment]
Many thx for your help, i'll try it now.

**Comment by AKASH2907** [comment]
I'm having the same issues. coral loss->0 classification loss->0.
Downloaded the dataset and placed it in office_31 folder.

Printing the dataloader for source, target_train and target_test looks like this:
Dataset ImageFolder
    Number of datapoints: 2817
    Root location: ../office_31/amazon
    StandardTransform
Transform: Compose(
               Resize(size=[256, 256], interpolation=PIL.Image.BILINEAR)
               RandomCrop(size=(224, 224), padding=None)
               RandomHorizontalFlip(p=0.5)
               ToTensor()
           )
Dataset ImageFolder
    Number of datapoints: 795
    Root location: ../office_31/webcam
    StandardTransform
Transform: Compose(
               Resize(size=[224, 224], interpolation=PIL.Image.BILINEAR)
               ToTensor()
           )
Dataset ImageFolder
    Number of datapoints: 795
    Root location: ../office_31/webcam
    StandardTransform
Transform: Compose(
               Resize(size=[224, 224], interpolation=PIL.Image.BILINEAR)
               ToTensor()
           )

Is there any problem you can point out?

**Comment by arnancjlx** [comment]
~office31/webcam/images/31classes?  Is your file path like this?
I think you should place your files like this:~office31/webcam/31classes

**Comment by AKASH2907** [comment]
Yeah, it worked. Thanks a ton.

**Comment by arnancjlx** [comment]
How about your coral_loss value now? I used the updated code and just got a pretty small value, like 0.0003.

**Comment by AKASH2907** [comment]
Yeah, I checked. I was just gonna commenting on that now. The coral loss is way too small. 0.00001 to 0.000003. Is it meant to be? @jindongwang

**Comment by AKASH2907** [comment]
I also want to know that at the time training in models.py. Model input is as 
model(source, target)

Whereas at the inference time, the model input is
s_output, _ = model(target_test_loader, None)

Shouldn't the target_test_loader to be passes in the second variable?

**Comment by AKASH2907** [comment]
When we load the target_train_loader & target_test_loader they have same number of images?
Isn't the number of datapoints should be different for training and testing? @jindongwang

**Comment by jindongwang** [comment]
Sorry for the late reply. I'm also wondering about the effect of this CORAL loss. However, I have checked the code and make sure it is the right implementation of DCORAL. I still couldn't find out why.

Currently, I suggest you increase the batch size to make it larger (i.e. 128, 256) since the CORAL is a second-order statistics of the features. In this way, you can see if it still has the same problem.

Of course, you are encouraged to check my code to see where I got wrong. Keep in touch.

**Comment by jindongwang** [comment]
@AKASH2907 @arnancjlx I rewrite the code of DCORAL and DDC, please check it again to see if your problem still happens. I'm considering closing this issue.

**Comment by arnancjlx** [comment]
@jindongwang I've rerun the updated code and the problem has been solved, thanks!
